### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by EuhenySiniy**

### CODE REVIEW

File: src/car/app/Car.java [hunk 1]
```diff
@@ -1,4 +1,116 @@
 package car.app;
 
+import java.util.ArrayList;
+
 public class Car {
+    private final int yearOfManufacture;
```
Comment (Anderwerty) [inline src/car/app/Car.java]: Don't use type int or String for the date. It is better  LocalDate.

File: src/car/app/Car.java [hunk 2]
```diff
@@ -1,4 +1,116 @@
...
+    private int passenger;
+    private int speed;
+    private ArrayList<CarWheel> wheels;
+    private ArrayList<CarDoor> doors;
+
+    public Car(int yearOfManufacture) {
+        this.yearOfManufacture = yearOfManufacture;
+    }
+
+    public Car(int yearOfManufacture, String typeMotor, int maxSpeed, double accelerationTo100, int passengerCapacity, int passenger, int speed) {
```
Comment (Anderwerty) [inline src/car/app/Car.java]: too long line.

File: src/car/app/Car.java [hunk 3]
```diff
@@ -1,4 +1,116 @@
...
+            this.passenger = 0;
+        } else {
+            return;
+        }
+    }
+
+    public CarDoor getDoor(int indx) {
+        try {
+            return doors.get(indx);
+        } catch (ArrayIndexOutOfBoundsException e) {
```
Comment (Anderwerty) [inline src/car/app/Car.java]: To write logic on exception is not good practice.

File: src/car/app/Car.java [hunk 4]
```diff
@@ -1,4 +1,116 @@
...
+            return null;
+        }
+    }
+
+    public CarWheel getWheel(int indx) {
+        try {
+            return wheels.get(indx);
```
Comment (Anderwerty) [inline src/car/app/Car.java]: The same

File: src/car/app/Car.java [hunk 5]
```diff
@@ -1,4 +1,116 @@
...
+            return null;
+        }
+    }
+
+    public void removeTheWheels() {
+        wheels.clear();
+    }
+
+    public void addTheWheel(int x) {
+        for (int a = 0; a < x; a++) {
```
Comment (Anderwerty) [inline src/car/app/Car.java]: `a` is not the best name for the iterable variable.  Better: n,m,i,k,j.

File: src/car/app/CarDoor.java [hunk 6]
```diff
@@ -1,4 +1,60 @@
 package car.app;
 
 public class CarDoor {
+    private boolean isDoor;
+    private boolean isWindow;
+
+    public CarDoor() {
+
```
Comment (Anderwerty) [inline src/car/app/CarDoor.java]: redundant blank line

File: src/car/app/CarDoor.java [hunk 7]
```diff
@@ -1,4 +1,60 @@
...
+    }
+
+    public CarDoor(boolean isDoor, boolean isWindow) {
+        this.isDoor = isDoor;
+        this.isWindow = isWindow;
+    }
+
+    public boolean isDoor() {
+        return isDoor();
```
Comment (Anderwerty) [inline src/car/app/CarDoor.java]: Infinite recursion

File: src/car/app/CarDoor.java [hunk 8]
```diff
@@ -1,4 +1,60 @@
...
+
+    public void closeWindow() {
+        this.isWindow = false;
+    }
+
+    public void closeOpenWindow() {
+        this.isWindow = !this.isWindow;
+    }
+
+    public void info() {
```
Comment (Anderwerty) [inline src/car/app/CarDoor.java]: Use constants for these phrases.

File: src/car/app/CarDoor.java [hunk 9]
```diff
@@ -1,4 +1,60 @@
...
+        if (isWindow = true) {
```
Comment (Anderwerty) [inline src/car/app/CarDoor.java]: This is not a comparison, it is an assignment. Try calling this method twice. Look at the result.

File: src/car/app/CarDoor.java [hunk 10]
```diff
@@ -1,4 +1,60 @@
...
+            System.out.println("Window open");
+        } else {
+            System.out.println("Window closed");
+        }
+        if (isDoor = true) {
```
Comment (Anderwerty) [inline src/car/app/CarDoor.java]: The same. Also, try to think about how to use the ternary operator here.

File: src/car/app/CarWheel.java [hunk 11]
```diff
@@ -1,4 +1,29 @@
 package car.app;
 
 public class CarWheel {
+    private double wheelWear;
+
+    public CarWheel() {
+        this.wheelWear = 1;
+    }
+
+    public CarWheel(double wheelWear) {
```
Comment (Anderwerty) [inline src/car/app/CarWheel.java]: Is possible to make wheelWear =100.

File: src/car/app/CarWheel.java [hunk 12]
```diff
@@ -1,4 +1,29 @@
...
+        this.wheelWear = wheelWear;
+    }
+
+    public void wheelChange() {
+        wheelWear = 1;
+    }
+
+    public void wipeWheel(double percent) {
```
Comment (Anderwerty) [inline src/car/app/CarWheel.java]: Could percent be less than 0, or bigger than 100?

**Comment by EuhenySiniy** [comment]
> Fix comments.

All comments has been fixed.

File: src/car/app/Car.java [hunk 13]
```diff
@@ -1,4 +1,123 @@
 package car.app;
 
+import java.util.ArrayList;
+import java.util.Date;
```
Comment (Anderwerty) [inline src/car/app/Car.java]: Don't use Date. It is an old class. Better LocalDate

File: src/car/app/Car.java (hunk 4 above)
Comment (Anderwerty) [reply src/car/app/Car.java]: what if index>wheels.size()?

File: src/car/app/Car.java [hunk 14]
```diff
@@ -1,4 +1,123 @@
...
+
+    public CarDoor getDoor(int indx) {
+        if (indx > 0) {
+            return doors.get(indx);
+        } else {
+            return null;
+        }
+    }
+
+    public CarWheel getWheel(int indx) {
```
Comment (Anderwerty) [inline src/car/app/Car.java]: typo.should be index

File: src/car/app/CarDoor.java [hunk 15]
```diff
@@ -1,4 +1,52 @@
 package car.app;
 
 public class CarDoor {
+    private boolean isDoor;
+    private boolean isWindow;
+
+    public CarDoor() {
+    }
+
+    public CarDoor(boolean isDoor,
```
Comment (Anderwerty) [inline src/car/app/CarDoor.java]: Better CarDoor(boolean isDoor,boolean isWindow) {

File: src/car/app/Car.java [hunk 16]
```diff
@@ -1,4 +1,123 @@
...
+    private int passenger;
+    private int speed;
+    private ArrayList<CarWheel> wheels;
+    private ArrayList<CarDoor> doors;
+
+    public Car(Date yearOfManufacture) {
+        this.yearOfManufacture = yearOfManufacture;
+    }
+
+    public Car(Date yearOfManufacture,
```
Comment (Anderwerty) [inline src/car/app/Car.java]: Better
String typeMotor,
 public Car(Date yearOfManufacture, int maxSpeed, double accelerationTo100,
               int passengerCapacity, int passenger, int speed) {

File: src/car/app/Car.java [hunk 17]
```diff
@@ -1,4 +1,123 @@
...
+
+    public void getOutPassanger() {
+        if (passenger > 0) {
+            passenger--;
+        } else {
+            return;
+        }
+    }
+
+    public void getOutPassangers() {
```
Comment (Anderwerty) [inline src/car/app/Car.java]:
``` else {
            return;
        }```
Redundant code.

File: src/car/app/Car.java [hunk 18]
```diff
@@ -1,4 +1,123 @@
...
+        if (passenger > 0) {
+            this.passenger = 0;
+        } else {
+            return;
+        }
+    }
+
+    public CarDoor getDoor(int indx) {
```
Comment (Anderwerty) [inline src/car/app/Car.java]: typo

File: src/car/app/Car.java [hunk 19]
```diff
@@ -1,4 +1,110 @@
...
+        this.passengerCapacity = passengerCapacity;
+        this.passenger = passenger;
+        this.speed = speed;
+    }
+
+    public void changeSpeed(int speed) {
+        this.speed = speed;
+    }
+
+    public void newPassanger() {
```
Comment (Anderwerty) [inline src/car/app/Car.java]:
```
else {
            return;
        }
```
redundant code.

File: src/car/app/Car.java [hunk 20]
```diff
@@ -1,4 +1,110 @@
...
+        if (passenger <= passengerCapacity) {
+            passenger++;
+        } else {
```
Comment (Anderwerty) [inline src/car/app/Car.java]: The same.

File: src/car/app/Car.java [hunk 21]
```diff
@@ -1,4 +1,110 @@
...
+            return;
+        }
+    }
+
+    public void getOutPassanger() {
+        if (passenger > 0) {
+            passenger--;
+        } else {
```
Comment (Anderwerty) [inline src/car/app/Car.java]: the same

**Pull request merged**